common/app/common/dfp/TagSponsorship.scala

```diff
@@ -100,10 +83,18 @@ case class HighMerchandisingLineItem(
   ) {
   val customTargets = customTargetSet.map(_.targets)
   val editions = customTargets.flatMap(sequence => sequence.filter((target) => target.name == "edition")).map(target => target.values)
-}
 
-case class HighMerchandisingLineItems(items: Seq[HighMerchandisingLineItem]) {
-  val sortedItems = items.sortBy(_.name)
+
+  def matchesAdUnitAndTag (adUnitSuffix: String, pageTags:Seq[Tag]): Boolean = {
+
+    def tagNames = pageTags map (_.name) map (_.replaceAll(" ","-").toLowerCase)
+
+    def matchesTag: Boolean = tagNames.exists(tags.contains)
+
+    def matchesAdUnit: Boolean = adUnits.exists(_.path contains adUnitSuffix)
```

Comment:
Yes, in general.  
You could argue in this case that `matchesAdUnit` doesn't need to be evaluated if `matchesTag` is false so it could be a `lazy val` instead.  But the benefit would be very small.